### PR TITLE
Different pool, different ConnectionManager

### PR DESCRIPTION
**Pull request by justmobilize**
Update ConnectionManager, to support multiple different pools at the same time.

Testing steps:
Have a device with onboard WiFi and either a ESP32SPI or WIZNET5K
Connect both and make requests on both and see that they are from different devices.

You can also use a WIZNET5K create it's request session first, and then disconnect it and verify the other device still works.

### CODE REVIEW

File: tests/mocket.py [hunk 1]
```diff
@@ -10,6 +10,13 @@
 MOCK_POOL_IP = "10.10.10.10"
 MOCK_HOST_1 = "wifitest.adafruit.com"
 MOCK_HOST_2 = "wifitest2.adafruit.com"
+MOCK_PATH_1 = "/testwifi/index.html"
```
Comment (justmobilize) [inline tests/mocket.py:13]: Updating to match what's in Requests

**Comment by justmobilize** [comment]
@anecdata here's another one that would be awesome if you could test (with everything else in main)

**Comment by justmobilize** [comment]
Since you easily have the ability to see where requests are coming from

**Review by anecdata — COMMENTED** [review]
Looks good! Test code and hardware photo [here](https://gist.github.com/anecdata/99148a1f2f779140d9e3b6c84006be37).

`Adafruit CircuitPython 9.1.0-beta.1 on 2024-04-19; FeatherS3 with ESP32S3`
(all other libraries standard latest except for this PR #8 version of ConnectionManager)

```
native wifi: UnexpectedMaker FeatherS3
ethernet:    WIZnet 850io (W5500) "BFF"
esp32spi:    Adafruit Airlift Breakout
```

Let it loop for a while, and verified that the router view of the HTTP sessions / IPv4 addresses matched the serial print sequence.

**Comment by anecdata** [comment]
Looks good from a testing perspective, maybe someone wants to code review.

**Comment by justmobilize** [comment]
@dhalbert once this is merged, I can take care of any conflicts and open up the 3x `use new socketpool` PRs

**Review by dhalbert — APPROVED** [review]
lgtm!

**Comment by dhalbert** [comment]
@justmobilize is this a bugfix version bump, or minor or major version bump, in terms of behavior?

**Comment by justmobilize** [comment]
@dhalbert I would consider it `minor`. It shouldn't break anything as it's only changing how pseudo private data is stored.

**Pull request merged**